From ffd93634426c16482b8635a6601bc63271b3aa27 Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Mon, 5 Feb 2018 09:49:02 +0800
Subject: [PATCH 3410/5242] MLK-17523 drm/imx: dcss: do dec400d shadow trigger
 only for primary plane

commit  fada1d053c7d3c82b195e99d55bef8452715627c from
https://source.codeaurora.org/external/imx/linux-imx.git

Since only primary plane has DEC400D attached to it, the
shadow register trigger for DEC400D is only necessary to
be done for primary plane update.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/imx/dcss/dcss-plane.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/imx/dcss/dcss-plane.c b/drivers/gpu/drm/imx/dcss/dcss-plane.c
index eb6e535..338d375 100644
--- a/drivers/gpu/drm/imx/dcss/dcss-plane.c
+++ b/drivers/gpu/drm/imx/dcss/dcss-plane.c
@@ -377,7 +377,8 @@ static void dcss_plane_atomic_update(struct drm_plane *plane,
 					   pixel_format, dcss_plane->alpha_val,
 					   dcss_plane->use_global_val)) {
 		dcss_plane_atomic_set_base(dcss_plane);
-		dcss_dec400d_shadow_trig(dcss_plane->dcss);
+		if (plane->type == DRM_PLANE_TYPE_PRIMARY)
+			dcss_dec400d_shadow_trig(dcss_plane->dcss);
 		return;
 	}
 
-- 
1.7.9.5

